packages/mdc-list/foundation.js

```diff
@@ -371,6 +412,38 @@ class MDCListFoundation extends MDCFoundation {
   hasCheckboxOrRadioAtIndex_(index) {
     return this.adapter_.hasCheckboxAtIndex(index) || this.adapter_.hasRadioAtIndex(index);
   }
+
+  /**
+   * @param {!Index} index
+   * @return {boolean}
+   */
+  isIndexValid_(index) {
+    const listSize = this.adapter_.getListItemCount();
+
+    if (index instanceof Array) {
+      if (!this.hasCheckboxAtIndex_(index)) {
+        throw new Error('MDCListFoundation: Array of index is only supported for checkbox based list');
+      }
+      return index.some((i) => this.isIndexInRange_(i));
+    } else {
+      return index >= 0 && index < listSize;
```

Comment:
Should we be checking that there is _not_ a checkbox at the index here?

Currently if you pass a single number (not an array) in the case of a checkbox list, it checks the indicated index _in addition_ to what is already checked... but then `list.selectedIndex` reports just that index, not all of the selected indices. I think it'd be safer to prevent this behavior (and insist that set/get are symmetrical) than try to solve for it.

I suspect this logic is also being depended upon for clicks/keydowns. Clearly if I interact with a checkbox list to select an item, `selectedIndex` then reports only that item regardless of what else is already selected, which matches the behavior I'm describing here. However, I can also uncheck via list item interaction, in which case `selectedIndex` _still_ reports the last interacted item - even though it's now _deselected_.

Reply:
`selectedIndex` is now synced with keyboard / mouse interacted selected state change. Thanks for spotting this out.

Also, Checkbox based list only accepts array of indexes.